src/common/components/Recaptcha/Recaptcha.tsx

```diff
@@ -71,50 +71,50 @@ export const Recaptcha = ({
   }, [grecaptcha, buttonId])
 
   const handleReCaptchaVerify = useCallback(async () => {
-    // if (!ready) {
-    //   $log.info('Execute recaptcha not yet available')
-    //   return
-    // }
-
-    // grecaptcha.ready(async () => {
-    //   try {
-    //     const token = await grecaptcha.execute(recaptchaKey, { action: action })
-    //     const serverResponse = await fetch(`${apiUrl}/recaptcha/verify`, {
-    //       method: 'POST',
-    //       body: JSON.stringify({ token: token }),
-    //       headers: { 'Content-Type': 'application/json' }
-    //     })
-    //     const data = await serverResponse.json()
-
-    //     if (data?.success === true && isValidAction(data?.action)) {
-    //       if (data.score < 0.5) {
-    //         $log.warn(`Captcha score was below the threshold: ${data.score}`)
-    //         onFailure(componentInstance)
-    //         return
-    //       }
-    //       onSuccess(componentInstance)
-    //       return
-    //     }
-    //     if (data?.success === false && isValidAction(data?.action)) {
-    //       $log.warn('Recaptcha call was unsuccessful, continuing anyway')
-    //       onSuccess(componentInstance)
-    //       return
-    //     }
-    //     if (!data) {
-    //       $log.warn('Data was missing!')
-    //       onSuccess(componentInstance)
-    //       return
-    //     }
-    //     if (!isValidAction(data?.action)) {
-    //       $log.warn(`Invalid action: ${data?.action}`)
-    //       onFailure(componentInstance)
-    //     }
-    //   } catch (error) {
-    //     $log.error(`Failed to verify recaptcha, continuing on: ${error}`)
-    //     onSuccess(componentInstance)
-    //   }
-    // })
-    onSuccess(componentInstance)
+    if (!ready) {
+      $log.info('Execute recaptcha not yet available')
+      return
+    }
+
+    grecaptcha.ready(async () => {
+      try {
+        const token = await grecaptcha.execute(recaptchaKey, { action: action })
+        const serverResponse = await fetch(`${apiUrl}/recaptcha/verify`, {
+          method: 'POST',
+          body: JSON.stringify({ token: token }),
+          headers: { 'Content-Type': 'application/json' }
+        })
+        const data = await serverResponse.json()
+
+        if (!data || !data.score || !data.action) {
```

Comment:
Are we positive that the action was returning `undefined` and not `"undefined"` (as a string)?

Reply:
I see. `action` itself is undefined.

Reply:
I'm about 90% sure; I think it was something like `{ message: "Over Enterprise free quota. Please ensure your project has an active billing account" }` as a response.